### PR TITLE
Switch to Toolbar

### PR DESCRIPTION
**Pull request by ByteHamster**
This allows to show the sidebar above the ActionBar, giving more space for folders.

### CODE REVIEW

**Comment by cketti** [comment]
This doesn't play well with the changes from #3771. `K9PreferenceActivity` needs a theme with action bar.

**Comment by ByteHamster** [comment]
Do you have a suggestion what should be done instead?

**Comment by ByteHamster** [comment]
@cketti what do you think about adding new themes without an ActionBar and then adding a method that converts the R constant from the normal theme to the one without ActionBar? Then the theme without ActionBar can be applied only to MessageList.

This probably also reduces the diff but I am not sure if it is "clean" enough for you to convert themes.

**Comment by cketti** [comment]
I like that you converted all of the activities to use toolbars. I think we should keep that, then add "legacy themes" with an action bar. Since they will only be used by `K9PreferenceActivity` subclasses we can omit most of the attributes of the regular K-9 themes.

To avoid having to call `setSupportActionBar()` in all activities I suggest we add a method `K9Activity.setLayout(int layoutResId)` that calls `setContentView()` and then `setSupportActionBar()`.

File: app/ui/src/main/res/layout/about.xml [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
-<WebView xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/about_view"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent" />
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+              android:orientation="vertical"
+              android:layout_height="fill_parent"
+              android:layout_width="fill_parent">
+
+    <android.support.v7.widget.Toolbar
```
Comment (williamvds) [inline app/ui/src/main/res/layout/about.xml]: The same toolbar element appears to be used in every layout that needs one  
How about pulling this out into its own layout, then using `<include layout="@layout/toolbar.xml"/>` here instead?

**Comment by cketti** [comment]
There are lots of subclasses of `K9Activity` that haven't been updated yet and are now missing an actionbar.

Using a custom view with a toolbar doesn't seem to automatically disable the regular actionbar title. Adding `actionBar.setDisplayShowTitleEnabled(false);` in `MessageList` fixed the issue for me.

**Comment by ByteHamster** [comment]
I hope that I have finally found all child activities. Sorry for the noise.

I verified that by clicking each activity in the list and searching for "setContentView". Do you think that is enough?

screenshot_20181204_221630

**Comment by cketti** [comment]
I believe you found all activities. However, the following layouts have small display issues:
- `account_setup_account_type`: padding on root container which also affects toolbar
- `account_setup_options`: layout contains two toolbars
- `crypto_key_transfer`: padding on root container which also affects toolbar
- `upgrade_databases`: gravity attribute on root container which also affects toolbar

And toolbar doesn't seem to support elevation on API levels prior to 21. That's an issue for another pull request, though.

**Comment by cketti** [comment]
Thanks!

**Comment by ByteHamster** [comment]
Sorry for ignoring your review. I was busy with university

**Comment by cketti** [comment]
No worries. You already did most of the work. I just added some minor cleanup. Thanks again!

**Pull request merged**